### ISSUE TITLE
[Feat]: Support alternate compression algorithms for dbengine storage on-disk.

### ISSUE THREAD

**Issue by Ferroin** [netdata/netdata#14128]
### Problem

Currently, the dbengine only support either not compressing data on disk, or using LZ4 in fast mode for storage on-disk. This makes sense for nodes that need to provide data in real-time, but limits the fine-tuning possible in this area.

Not having more space efficient compression options available limits the practicality of things like archival storage of metrics internal to Netdata.

### Description

Add additional compression algorithms as options for the dbengine’s on-disk storage.

### Importance

nice to have

### Value proposition

1. Users who wish to archive old data internally to Netdata (as opposed to using an exporter and external storage) can do so in a way that provides more efficient disk space usage.
2. Users who are dealing with systems with fast CPUs but slow storage can optimize for better access times by using higher compression.

### Proposed implementation

A simple initial implementation would be to add LZ4HC support. This is a slightly slower version of the LZ4 compressor that provides better compression ratios in many cases, but still decompresses just as fast. Given that we already depend on LZ4 for the dbengine, this should be easy to add quickly without needing any new dependencies.

For a more comprehensive implementation, I recommend adding support for [zstd](http://facebook.github.io/zstd/) compression, with options to control the compression level. Using the default parameters, zstd gets compression ratios roughly equivalent to zlib’s default parameters, but with a compression and decompression speed ranging between 3-6 times faster. It additionally provides a specialized trainable dictionary compression mode that can provide tailored compression for ‘small’ data (on the order of a few kB per item) which can easily achieve compression ratios in excess of 8.0 for such data if it’s highly repetitive, which seems ideally suited to our usage.

**Comment by ilyam8** [comment]
ZSTD compression support was added in #17244
